apps/hashdotai/blog/039_self-building-database.mdx

```diff
@@ -0,0 +1,128 @@
+- Extracting important relevant information automatically from never-ending product changelogs, company blogs, press releases, social media posts and other online publications
+
+## Identifying opportunities
+
+No matter what side of an opportunity you're on, automating the legwork of identifying opportunities and qualifying/filtering them can be time-consuming. Whether you're...
+
+- Prospecting for investment opportunities, or seeking out potential investors;
+- Headhunting potential new recruits, or researching potential open roles;
+- Identifying and qualifying sales targets, or finding the best solution provider for your needs;
+- Selling commercial real estate, or looking for a unique space;
+
+...you're typically going to spend a large amount of time Googling, clicking through onto a variety of different websites and results, grappling with inadequate search filters, and manually doing a lot of the information parsing yourself.
+
+## Enabling complex systems modeling
+
+Complex systems can be incredibly sensitive to their initial conditions. Even small differences between a variable used in a model, and the actual real-world value of a thing, can cause huge differences in simulated outcomes.
+
+In many domains, ensuring models are as accurate as possible requires the observation and collection of accurate, real-time, real-world information. Doyne Former, of the Santa Fe Institute and Oxford's Institute for New Economic Thinking, refers to this as the need for "collective awareness". Using publicly available unstructured data, we can create live, composite metrics (e.g. gauges of price volatility and inflation, with respect to different types of products) and incorporate these into simulations (in this example, economic ones). Doing this type of thing with a self-building database becomes immensely easier.
+
+_n.b. For readers of this post who don't know our origins, HASH was founded in 2019 as a multi-agent systems research lab, building [agent-based simulation modeling](/cases/simulation) tools._
+
+# Technical requirements
+
+Given the database that we "imagine", there are quite a few technical components that need to exist to make it real. Some of these are easy to find, while others do not obviously already exist.
+
+## A type system
+
+Many people have tried to build graphs of "linked open data" before, but they haven't worked, or at least gained the dominance, because different people conceptualize the same things differently. Not all folks agree on defining things the same way, often for very good reasons (in spite of the best efforts of projects like _schema.org_), and even when _they do_, individuals often care about different aspects of those things. For example, one person might care about how scenic a given walking route is, while a second might care about how hilly it is, while a third is concerned with its directness and efficiency.
```

Comment:
```suggestion
Many people have tried to build graphs of "linked open data" before, but they haven't worked, or at least gained dominance, because different people conceptualize the same things differently. Not all folks agree on defining things the same way, often for very good reasons (in spite of the best efforts of projects like _schema.org_), and even when _they do_, individuals often care about different aspects of those things. For example, one person might care about how scenic a given walking route is, while a second might care about how hilly it is, while a third is concerned with its directness and efficiency.
```
```suggestion
Many people have tried to build graphs of "linked open data" before, but they haven't worked, or at least gained the dominance required to be a truly open graph rather than a series of bespoke silos, because different people conceptualize the same things differently. Not all folks agree on defining things the same way, often for very good reasons (in spite of the best efforts of projects like _schema.org_), and even when _they do_, individuals often care about different aspects of those things. For example, one person might care about how scenic a given walking route is, while a second might care about how hilly it is, while a third is concerned with its directness and efficiency.
```